### PR TITLE
[ENH] Link to filename element definitions in filename templates

### PR DESCRIPTION
**Pull request by tsalo**
Closes None, but will help with #612.

Changes proposed:

- Filename elements (e.g., suffixes, entities, extensions, datatypes) in the filename templates throughout the specification now link to the definitions of those elements. For example, if a filename allows the `task` entity, that portion of the filename template will link to the `task` entity definition in the Entities appendix page.

~~This feels really close. The problem is that the paths aren't working properly.~~

~~Basically, hrefs like `SPEC_ROOT/99-appendices/09-entities.md#subject` are resolving to https://bids-specification--1228.org.readthedocs.build/en/1228/04-modality-specific-files/99-appendices/09-entities.md#subject. Note the `04-modality-specific-files`.~~

### CODE REVIEW

**Review by Remi-Gau — COMMENTED** [review]
OMG. This looks amazing already.

Definitely worth trying to make this last bit work I think (no pressure)

**Comment by tsalo** [comment]
Yes, ultimately I'd like to link to the (1) entities, (2) datatypes, (3) entity value formats (index vs. label vs. enum), (4) suffixes, and (5) extensions. Adding in new links won't be too hard, so once I actually have the links working I'll add a whole bunch of other ones.

File: tools/schemacode/bidsschematools/render.py [hunk 1]
```diff
@@ -205,11 +208,16 @@ def make_filename_template(schema, n_dupes_to_combine=6, **kwargs):
 
     paragraph = ""
     # Parent directories
-    paragraph += "{}-<{}>/\n\t[{}-<{}>/]\n".format(
-        schema["objects"]["entities"]["subject"]["name"],
-        schema["objects"]["entities"]["subject"]["format"],
-        schema["objects"]["entities"]["session"]["name"],
-        schema["objects"]["entities"]["session"]["format"],
+    paragraph += (
+        '<a href="SPEC_ROOT/99-appendices/09-entities.md#subject">'
```
Comment (effigies) [inline tools/schemacode/bidsschematools/render.py:212]: Don't you just want things like:

```suggestion
        '<a href="/99-appendices/09-entities.html#subject">'
```

Comment (tsalo) [reply tools/schemacode/bidsschematools/render.py]: That works, thanks! I assumed we _didn't_ want to reference HTML paths or use absolute paths, but if this isn't the preferred approach we can always revisit it later.

**Comment by tsalo** [comment]
It seems to be working, and it is GLORIOUS!!

**Comment by sappelhoff** [comment]
amazing :clap: :rocket:

**Comment by tsalo** [comment]
Oh but now the PDF looks awful. 😞 

image

**Comment by sappelhoff** [comment]
can you selectively switch this behavior off for the pdf build?

**Comment by tsalo** [comment]
@sappelhoff I have refactored things a bit to toggle the formatting based on a new function argument (`pdf_format`). I need to figure out how to get the pdf build to use this new argument now.

**Comment by tsalo** [comment]
Okay, everything's working now. I will probably need to improve the tests to cover `pdf_format=False` (i.e., HTML outputs), and docstrings definitely need to be improved, but other than that it's wonderful!

File: tools/schemacode/bidsschematools/render.py [hunk 2]
```diff
@@ -257,36 +327,86 @@ def make_filename_template(schema, n_dupes_to_combine=6, **kwargs):
...
             full_strings = sorted(full_strings)
             if full_strings:
                 paragraph += "\n".join(full_strings) + "\n"
 
     paragraph = paragraph.rstrip()
-    codeblock = "Template:\n```Text\n" + paragraph + "\n```"
+    if pdf_format:
+        codeblock = f"Template:\n```Text\n{paragraph}\n```"
+    else:
+        codeblock = f'Template:\n<pre style="overflow-x:scroll;">\n{paragraph}\n</pre>'
```
Comment (effigies) [inline tools/schemacode/bidsschematools/render.py:405]: We lost the gray background.

```suggestion
        codeblock = f'Template:\n<pre style="overflow-x:scroll;"><code>\n{paragraph}\n</code></pre>'
```

Comment (tsalo) [reply tools/schemacode/bidsschematools/render.py:405]: Good catch! I didn't even notice that.

**Review by Remi-Gau — APPROVED** [review]
Great work @tsalo

File: tools/schemacode/bidsschematools/render.py [hunk 3]
```diff
@@ -318,36 +391,88 @@ def make_filename_template(schema, n_dupes_to_combine=6, **kwargs):
...
             if full_strings:
                 paragraph += "\n".join(full_strings) + "\n"
 
     paragraph = paragraph.rstrip()
-    codeblock = "Template:\n```Text\n" + paragraph + "\n```"
+    if pdf_format:
+        codeblock = f"Template:\n```Text\n{paragraph}\n```"
+    else:
+        codeblock = (
+            f'Template:\n<pre style="overflow-x:scroll;"><code>\n{paragraph}\n</code></pre>'
```
Comment (effigies) [inline tools/schemacode/bidsschematools/render.py:470]: Tested locally. Fixes an extra line at the top of each block and the doubled-up scrollbar.

```suggestion
            f'Template:\n<div class="highlight"><pre><code>{paragraph}\n</code></pre></div>'
```

## Before

Screenshot from 2022-08-22 16-31-56

## After
Screenshot from 2022-08-22 16-31-50

Comment (tsalo) [reply tools/schemacode/bidsschematools/render.py]: Thanks @effigies!

**Review by effigies — APPROVED** [review]

**Review by sappelhoff — APPROVED** [review]

**Comment by sappelhoff** [comment]
Thanks @tsalo!

**Comment by tsalo** [comment]
Uh oh, it looks like the new links don't point to the specific language and version. They were fine in the CI build, but I guess that this specific issue wasn't covered.

So instead of:

https://bids-specification.readthedocs.io/en/latest/99-appendices/09-entities.html#sub

Links go to:

https://bids-specification.readthedocs.io/99-appendices/09-entities.html#sub

**Comment by tsalo** [comment]
This seems to be the relevant part of the mkdocs documentation: https://www.mkdocs.org/user-guide/writing-your-docs/#linking-from-raw-html. Unfortunately, I still don't know how to "manually format the link appropriately".

**Comment by effigies** [comment]
We can't. We'll need to figure out the SPEC_ROOT stuff.

**Pull request merged**